### PR TITLE
Adding support to update OneToMany associations

### PR DESCRIPTION
**Pull request by scriptom**
As of now, we weren't able to update one-to-many associations in a POST call to an entity. This pull requests seeks to provide this functionality for all OneToMany, using Bullhorn's replaceAll syntax, that consists in an array containing the IDs that should replace all entities in an association.

### CODE REVIEW

**Review by taswartz — APPROVED** [review]
Really clean. I like that you refactored the `handleUpdateEntity` to be overloaded now too

File: src/test/java/com/bullhornsdk/data/api/helper/RestJsonConverterTest.java [hunk 1]
```diff
@@ -154,4 +159,28 @@ public void testStandardFileContentWithExtraProp() {
         assertEquals("SomeContent", file.getFileContent());
         assertEquals("FileName", file.getName());
     }
+
```
Comment (taswartz) [inline src/test/java/com/bullhornsdk/data/api/helper/RestJsonConverterTest.java:162]: Can we create another Test case that ensures that if the OneToMany relationship is completely null, that the JSONObject does not get populated with an empty replaceAll?

**Review by taswartz — CHANGES_REQUESTED** [review]
We may want to review this just to make sure we are not bypassing any unintended functionality

File: src/test/java/com/bullhornsdk/data/api/helper/RestJsonConverterTest.java (hunk 1 above)
Comment (scriptom) [reply src/test/java/com/bullhornsdk/data/api/helper/RestJsonConverterTest.java:162]: If the OneToMany relationship is null, it will be ignored by Jackson and won't be present in the resulting JSON.

**Review by johnsully83 — APPROVED** [review]

**Review by taswartz — APPROVED** [review]
